packages/amplify-environment-parameters/src/types.ts

```diff
@@ -0,0 +1,35 @@
+import { ResourceParameterManager } from './resource-parameter-manager';
+
+/**
+ * Env metadata type
+ */
+export type IEnvironmentMetadata = {
+  readonly AuthRoleName: string,
+  readonly AuthRoleArn: string,
+  readonly UnauthRoleArn: string,
+  readonly UnauthRoleName: string,
+  readonly Region: string,
+  readonly DeploymentBucketName: string,
+  readonly StackName: string,
+  readonly StackId: string,
+  AmplifyAppId: string,
+  PermissionsBoundaryPolicyArn: string | undefined,
+  write: (respectIsDirty?: boolean, writeTarget?: WriteTarget) => void
```

Comment:
what does `respectIsDirty` do ? - I'd rather add `forceWrite` API and not expose dirty flag in case internal implementation changes.
Have we consider keeping `IEnvironmentMetadata` pure data structure and have other type provide functionalities like write ? I.e. should `IEnvironmentParameterManager.write(metadata, target)` ?